test/e2e/service.go

```diff
@@ -251,28 +253,35 @@ var _ = Describe("Services", func() {
 		}
 		host := hosts[0]
 
+		By("verifying service1 is up")
 		expectNoError(verifyServeHostnameServiceUp(c, ns, host, podNames1, svc1IP, servicePort))
+
+		By("verifying service2 is up")
 		expectNoError(verifyServeHostnameServiceUp(c, ns, host, podNames2, svc2IP, servicePort))
 
 		// Stop service 1 and make sure it is gone.
+		By("stopping service1")
 		expectNoError(stopServeHostnameService(c, ns, "service1"))
 
+		By("verifying service1 is not up")
 		expectNoError(verifyServeHostnameServiceDown(c, host, svc1IP, servicePort))
+		By("verifying service2 is still up")
 		expectNoError(verifyServeHostnameServiceUp(c, ns, host, podNames2, svc2IP, servicePort))
 
 		// Start another service and verify both are up.
+		By("creating service3 in namespace " + ns)
 		podNames3, svc3IP, err := startServeHostnameService(c, ns, "service3", servicePort, numPods)
 		Expect(err).NotTo(HaveOccurred())
 
 		if svc2IP == svc3IP {
-			Failf("VIPs conflict: %v", svc2IP)
+			Failf("service IPs conflict: %v", svc2IP)
 		}
 
+		By("verifying service2 is still up")
 		expectNoError(verifyServeHostnameServiceUp(c, ns, host, podNames2, svc2IP, servicePort))
-		expectNoError(verifyServeHostnameServiceUp(c, ns, host, podNames3, svc3IP, servicePort))
 
-		expectNoError(stopServeHostnameService(c, ns, "service2"))
-		expectNoError(stopServeHostnameService(c, ns, "service3"))
```

Comment:
There's no point in manually removing the services - namespace deletion handles it.  We should, in general, remove cleanup code where it is not related to the test, and depend on the framework.